### DISCUSSION TITLE
[@types/oojs-ui] Types for window lifecycle promises are not supported

### DISCUSSION THREAD

**Discussion by jwbth** [DefinitelyTyped/DefinitelyTyped#69665] [Issues with a @types package]
Example:
```js
async function func() {
  await mw.loader.using('oojs-ui');

  const windowManager = new OO.ui.WindowManager();
  $(document.body).append(windowManager.$element);

  const dialog = new OO.ui.MessageDialog();
  windowManager.addWindows([dialog]);
  const instance = windowManager.openWindow(dialog, {
    message: 'Message',
    actions: [
      {
        label: 'Close',
        action: 'close',
      },
    ],
  });
  const { action } = await instance.closed;
  console.log(action);
}

func();
```
This code prints "close" to the console when the window is closed. But TypeScript doesn't understand what's going on:
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/33615628/c8a02103-a255-442c-a7a0-4e1514ffc9fd)

Using `.then()` instead doesn't make any difference.

Window lifecycle promises are documented at https://www.mediawiki.org/wiki/OOUI/Windows.

[a comment by an automated account is not shown]

**Comment by diskdance** [comment]
As I said on Discord, for a temporary solution `as` can be used, and eventually we should make `OO.ui.WindowsInstance` generic.
